cfme/tests/intelligence/chargeback/test_chargeback_long_term_rates.py

```diff
@@ -0,0 +1,442 @@
+    candu.enable_all()
+
+    yield
+
+    server_info.update_server_roles_db(original_roles)
+    candu.disable_all()
+
+
+@pytest.yield_fixture(scope="module")
+def assign_custom_rate(new_compute_rate):
+    """Assign custom Compute rate to the Enterprise and then queue the Chargeback report."""
+    description = new_compute_rate
+    enterprise = cb.Assign(
+        assign_to="The Enterprise",
+        selections={
+            'Enterprise': {'Rate': description}
+        })
+    enterprise.computeassign()
+    enterprise.storageassign()
+    logger.info('Assigning CUSTOM Compute rate')
+
+    yield
+
+    # Resetting the Chargeback rate assignment
+    enterprise = cb.Assign(
+        assign_to="The Enterprise",
+        selections={
+            'Enterprise': {'Rate': '<Nothing>'}
+        })
+    enterprise.computeassign()
+    enterprise.storageassign()
+
+
+def verify_records_rollups_table(appliance, provider):
+    """Verify that hourly rollups are present in the metric_rollups table."""
+    vm_name = provider.data['cap_and_util']['chargeback_vm']
+
+    ems = appliance.db.client['ext_management_systems']
+    rollups = appliance.db.client['metric_rollups']
+
+    with appliance.db.client.transaction:
+        result = (
+            appliance.db.client.session.query(rollups.id)
+            .join(ems, rollups.parent_ems_id == ems.id)
+            .filter(rollups.capture_interval_name == 'hourly', rollups.resource_name == vm_name,
+            ems.name == provider.name, rollups.timestamp >= date.today())
+        )
+
+    for record in appliance.db.client.session.query(rollups).filter(
+            rollups.id.in_(result.subquery())):
+        return any([record.cpu_usagemhz_rate_average,
+           record.cpu_usage_rate_average,
+           record.derived_memory_used,
+           record.net_usage_rate_average,
+           record.disk_usage_rate_average])
+    return False
+
+
+@pytest.fixture(scope="module")
+def resource_usage(vm_ownership, appliance, provider):
+    """Retrieve resource usage values from metric_rollups table."""
+    average_cpu_used_in_mhz = 0
+    average_memory_used_in_mb = 0
+    average_network_io = 0
+    average_disk_io = 0
+    average_storage_used = 0
+    consumed_hours = 0
+    vm_name = provider.data['cap_and_util']['chargeback_vm']
+
+    metrics = appliance.db.client['metrics']
+    rollups = appliance.db.client['metric_rollups']
+    ems = appliance.db.client['ext_management_systems']
+    logger.info('Deleting METRICS DATA from metrics and metric_rollups tables')
+
+    appliance.db.client.session.query(metrics).delete()
+    appliance.db.client.session.query(rollups).delete()
+
+    # Chargeback reporting is done on hourly and daily rollup values and not real-time values.So, we
+    # are capturing C&U data and forcing hourly rollups by running these commands through
+    # the Rails console.
+
+    def verify_records_metrics_table(appliance, provider):
```

Comment:
**REQUIRED**
Why did you put this function inside another function? Please take it out and place it near `verify_records_rollups_table`.